hbase-server/src/main/java/org/apache/hadoop/hbase/snapshot/SnapshotManifest.java

```diff
@@ -343,6 +343,13 @@ private void load() throws IOException {
     }
   }
 
+  /**
+   * Sets the status task for monitoring all the subtasks for Snapshot operation
+   */
+  public void setMonitoredTask(MonitoredTask statusTask) {
```

Comment:
Any reason to have a separate setter instead of passing it via create() ?

Reply:
Passing with create() sounds good, this class is anyways IA.Private, we can add new argument to public methods.

Reply:
There are few other places where  create method is getting called in src and test classes. I just thought this would be easier for the callers to set the status as optional when needed instead of passing in to the create call where everyone has pass in null explicitly in the absence of task object.

Reply:
It seems more straight to readers if we pass this in the `create()`. Also this class field can be final then. @sguggilam We can have two versions of `create()` I guess? The one supporting those call sites which do not need monitored task does not need to change.

Reply:
+1